tests/python/openassetio_test/test_PyRetainingSharedPtr.py

```diff
@@ -0,0 +1,203 @@
+#
+#   Copyright 2013-2021 The Foundry Visionmongers Ltd
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""
+Tests of C++ binding utilities, which require dummy C++ classes that
```

Comment:
[fix] We have a few cases where we know the value will have multiple `PyRetainingSharedPtrs` associated with it (eg: the return of `ManagerInterface::createState` being held in `Context.managerState`. Could we add a test where we check that lifetime is properly managed be independently constructed `PyRetainingSharedPtrs`?

Reply:
Added as part of 4e4c1fed - see `test_when_factory_feeds_container_and_container_destroyed_then_cpp_element_destroyed`